src/common/types.ts

```diff
@@ -72,7 +73,7 @@ function createPropertySignature(
 ) {
   return ts.factory.createPropertySignature(
     /*modifiers*/ undefined,
-    /*name*/ ts.factory.createIdentifier(name),
+    /*name*/ ts.factory.createIdentifier(addQuotesWhenHasDashes(name)),
```

Comment:
Nice! Very close but I think the AST way of doing this would be with a string literal instead of an identifier:

```ts
hasDashes(name) ? ts.factory.createStringLiteral(name) : ts.factory.createIdentifier(name)
```

Would you mind trying that please? 🙏

Reply:
Done